### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
The above build failed, because stupid mistake in unit tests, and a cancelled, almost finished successfully integraiton test build

**Review by itegulov — APPROVED** [review]

File: casper/src/main/scala/coop/rchain/casper/Validate.scala [hunk 1]
```diff
@@ -166,7 +165,7 @@ object Validate {
       true.pure[F]
     }
 
-  def version[F[_]: Applicative: Log](b: BlockMessage, version: Long): F[Boolean] = {
+  def version[F[_]: Monad: Log](b: BlockMessage, version: Long): F[Boolean] = {
```
Comment (JosephDenman) [inline casper/src/main/scala/coop/rchain/casper/Validate.scala:168]: Shouldn't this be `FlatMap` as it is in your change to `PacketHandler`?

**Review by JosephDenman — APPROVED** [review]
In one place you add a `Flatmap` constraint to `F[_]` to get `>>` and in another you add a `Monad` constraint. They should probably be consistent. Otherwise, LGTM.

**Comment by ArturGajowy** [comment]
Bors r+

File: casper/src/main/scala/coop/rchain/casper/Validate.scala (hunk 1 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/Validate.scala:168]: this me field needs applicative otherwise

**Pull request merged**